lib/src/client.dart

```diff
@@ -537,6 +539,37 @@ class Client {
     PaginationParams paginationParams = const PaginationParams(limit: 10),
     int messageLimit,
     bool onlyOffline = false,
+  }) {
+    final hash = base64.encode(utf8.encode(
+        '$filter${sort?.map((s) => s.toJson())}$options${paginationParams?.toJson()}$messageLimit$onlyOffline'));
```

Comment:
Consider extracting the map call to a private function 

Maybe:
```
String _asMap(sort){
  return sort?.map((s) => s.toJson());
}
```

Then:
```
final hash = base64.encode(utf8.encode(
   '$filter${_asMap(sort)}$options${paginationParams?.toJson()}$messageLimit$onlyOffline'));
```

It would help to make the string more readable.